torch/csrc/jit/pickler.cpp

```diff
@@ -159,25 +233,103 @@ void Pickler::pushString(const std::string& string) {
   stack_.insert(stack_.end(), string.begin(), string.end());
 }
 
-void Pickler::pushClass(PicklerClass cls) {
-  const auto& name = getClassName(cls);
-  // Write it to the tensor table
-  auto memo_entry = memo_map_.find(&name);
+void Pickler::pushGlobal(const std::string& name_temp) {
+  memoized_strings_.push_back(name_temp);
+  auto name = memoized_strings_.back();
+  auto memo_entry = memo_map_.find(&(memoized_strings_.back()));
```

Comment:
This doesn't seem right. What is the intention here? `&memoized_strings_.back()`is a pointer to a position in memoized_strings_. Since it was just inserted, it will never be in the memo_map_, and if the memoized_strings_ gets reallocated, then it will be a pointer to bogus data. memo_map_ only really works for reference IValue types. Memoizing non-ivalue strings will probably require a hash map from string -> memo id.